src/main/java/org/datadog/jenkins/plugins/datadog/DatadogGlobalConfiguration.java

```diff
@@ -488,14 +496,86 @@ public FormValidation doCheckTargetCredentialsApiKey(
         }
         return FormValidation.ok();
     }
 
+    /**
+     * Populates the targetCredentialsApplicationKey field from the configuration screen with all of the valid credentials
+     *
+     * @param item - The context within which to list available credentials
+     * @param targetCredentialsApplicationKey - A String containing the application key as a credential
+     * @return a ListBoxModel object used to display all of the available credentials.
+     */
+    public ListBoxModel doFillTargetCredentialsApplicationKeyItems(
+            @AncestorInPath Item item,
+            @QueryParameter("targetCredentialsApplicationKey") String targetCredentialsApplicationKey
+    ) {
+        StandardListBoxModel result = new StandardListBoxModel();
+        // If the users does not have permissions to list credentials, only list the current value
+        if (item == null) {
+            if (!Jenkins.get().hasPermission(Jenkins.ADMINISTER)) {
+                return result.includeCurrentValue(targetCredentialsApplicationKey);
+            }
+        } else {
+            if (!item.hasPermission(Item.EXTENDED_READ)
+                    && !item.hasPermission(CredentialsProvider.USE_ITEM)) {
+                return result.includeCurrentValue(targetCredentialsApplicationKey);
+            }
+        }
+        return result.includeEmptyValue()
+                .includeMatchingAs(ACL.SYSTEM,
+                        Jenkins.get(),
+                        StringCredentials.class,
+                        Collections.emptyList(),
+                        CredentialsMatchers.instanceOf(StringCredentials.class))
+                .includeCurrentValue(targetCredentialsApplicationKey);
+    }
+
+    /**
+     * Tests the targetCredentialsApplicationKey field from the configuration screen, to check its' validity.
+     *
+     * @param item - The context within which to list available credentials.
+     * @param targetCredentialsApplicationKey - A String containing the API key as a credential
+     * @return a FormValidation object used to display a message to the user on the configuration
+     * screen.
+     */
+    @RequirePOST
+    public FormValidation doCheckTargetCredentialsApplicationKey(
+            @AncestorInPath Item item,
+            @QueryParameter("targetCredentialsApplicationKey") String targetCredentialsApplicationKey
+    ) {
+        // Don't validate for users that do not have permission to list credentials
+        if (item == null) {
+            if (!Jenkins.get().hasPermission(Jenkins.ADMINISTER)) {
+                return FormValidation.ok();
+            }
+        } else {
+            if (!item.hasPermission(Item.EXTENDED_READ)
+                    && !item.hasPermission(CredentialsProvider.USE_ITEM)) {
+                return FormValidation.ok();
+            }
+        }
+        if (StringUtils.isBlank(targetCredentialsApplicationKey)) {
```

Comment:
The admin permissions are checked if there is no context available, otherwise "extended read" and "use item" permissions are checked: it works the same way as validation for the API key does

Reply:
Ah yes true, I was thinking of doTestConnection 🙇‍♀️